datasource/ethereum/src/ethereum_adapter.rs

```diff
@@ -840,144 +859,19 @@ where
-                    // in the block filter, transform the `block_filter` into
-                    // a `call_filter` and run `blocks_with_calls`
-                    let call_filter = EthereumCallFilter::from(block_filter);
-                    block_futs.push(Box::new(eth.blocks_with_calls(
-                        &logger,
-                        subgraph_metrics.clone(),
-                        from,
-                        to,
-                        call_filter,
-                    )));
-                }
-            }
-        }
-        Box::new(block_futs.concat2().and_then(|block_ptrs| {
-            let mut blocks = block_ptrs.into_iter().collect::<Vec<_>>();
-            blocks.sort_by(|a, b| a.number.cmp(&b.number));
-            future::ok(blocks)
-        }))
-    }
-
-    fn blocks(
-        &self,
-        logger: &Logger,
-        _: Arc<SubgraphEthRpcMetrics>,
-        from: u64,
-        to: u64,
-    ) -> Box<dyn Future<Item = Vec<EthereumBlockPointer>, Error = Error> + Send> {
-        let eth = self.clone();
-        let logger = logger.clone();
-
-        // Generate `EthereumBlockPointers` from `to` backwards to `from`
-        Box::new(
-            self.block_hash_by_block_number(&logger, to)
-                .map(move |block_hash_opt| EthereumBlockPointer {
-                    hash: block_hash_opt.unwrap(),
-                    number: to,
-                })
-                .and_then(move |block_pointer| {
-                    stream::unfold(block_pointer, move |descendant_block_pointer| {
-                        if descendant_block_pointer.number < from {
-                            return None;
-                        }
-                        // Populate the parent block pointer
-                        Some(
-                            eth.block_parent_hash(&logger, descendant_block_pointer.hash)
-                                .map(move |block_hash_opt| {
-                                    let parent_block_pointer = EthereumBlockPointer {
-                                        hash: block_hash_opt.unwrap(),
-                                        number: descendant_block_pointer.number - 1,
-                                    };
-                                    (descendant_block_pointer, parent_block_pointer)
-                                }),
-                        )
-                    })
-                    .collect()
-                })
-                .map(move |mut block_pointers| {
-                    block_pointers.reverse();
-                    block_pointers
-                }),
-        )
-    }
-
     fn blocks_with_logs(
         &self,
         logger: &Logger,
         subgraph_metrics: Arc<SubgraphEthRpcMetrics>,
         from: u64,
         to: u64,
         log_filter: EthereumLogFilter,
-    ) -> Box<dyn Future<Item = Vec<EthereumBlockPointer>, Error = Error> + Send> {
+    ) -> Box<dyn Future<Item = Vec<Log>, Error = Error> + Send> {
```

Comment:
Seems that the method name needs to change now, something like `logs_in_block_range`

Reply:
Yup

Reply:
The name does read weird, renamed it.